### PR TITLE
Add lookmovie

### PR DESCRIPTION
**Pull request by JipFr**
This pull request resolves #XXX

 - [x] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [ ] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [x] I have tested all of my changes.

### CODE REVIEW

File: src/providers/sources/lookmovie/video.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Config } from './type';
+
+export async function getVideoSources(config: Config): Promise<any> {
+  // Fetch video sources
+  let url = '';
+  if (config.type === 'show') {
+    url = `https://www.lookmovie2.to/api/v1/security/episode-access?id_episode=${config.episodeId}&hash=${config.hash}&expires=${config.expires}`;
+  } else if (config.type === 'movie') {
+    url = `https://www.lookmovie2.to/api/v1/security/movie-access?id_movie=${config.id_movie}&hash=${config.hash}&expires=${config.expires}`;
```
Comment (binaryoverload) [inline src/providers/sources/lookmovie/video.ts:9]: I assume the inconsistencies between `id_movie` and `episodeId` are LokLok's thing?

Also it seems that when the config option is passed in (https://github.com/movie-web/providers/pull/15/files#diff-20980de874adb0244ef7ce68a1e8e3b464304a20881a0183a425609b508ae7a0R58) it's using `movieId` not `id_movie`?

File: src/providers/sources/lookmovie/util.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { compareMedia } from '@/utils/compare';
+import { NotFoundError } from '@/utils/errors';
+
+import { Result } from './type';
+import { getVideoUrl } from './video';
+
+export async function searchAndFindMedia(media: MovieMedia | ShowMedia): Promise<Result | undefined> {
+  const searchRes = await fetch(
+    `https://lookmovie2.to/api/v1/${media.type}s/do-search/?q=${encodeURIComponent(media.title)}`,
+  ).then((d) => d.json());
```
Comment (mrjvs) [inline src/providers/sources/lookmovie/util.ts:13]: use the fetcher on the context. the fetcher can also handle querty params and base urls.
(dont use the proxied fetcher, as the streams are ip based)

Comment (mrjvs) [reply src/providers/sources/lookmovie/util.ts:13]: and it auto parses json

Comment (mrjvs) [reply src/providers/sources/lookmovie/util.ts:13]: (this applies to everywhere where fetch is used, ill only comment it once here)

File: src/providers/sources/lookmovie/video.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  return data;
+}
+
+export async function getVideoUrl(config: Config): Promise<string | null> {
+  // Get sources
+  const data = await getVideoSources(config);
+  const videoSources = data.streams;
+
+  // Find video URL and return it
+  const opts = ['1080p', '1080', '720p', '720', '480p', '480', 'auto'];
```
Comment (mrjvs) [inline src/providers/sources/lookmovie/video.ts:21]: I would prioritize auto, as I assume its a HLS stream of all qualities, just that the client can automatically select which one gets chosen.

File: src/providers/sources/lookmovie/index.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    embeds: [],
+    stream: {
+      playlist: videoUrl,
+      type: 'hls',
+      flags: [],
+    },
+  };
+}
+
+export const lookmovieScraper = makeSourcerer({
```
Comment (mrjvs) [inline src/providers/sources/lookmovie/index.ts:28]: make a new flag `IP_BASED`. which will mean that the streams are based on the ip it gets requested from.
then add the flag to both the streams and source itself

**Review by mrjvs — CHANGES_REQUESTED** [review]

**Comment by MemeCornucopia** [comment]
This would be a really good addition

**Comment by JipFr** [comment]
If it worked, sure. But it doesn't. They've re-added the captcha, so this provider will never work. I'm closing the PR until that might change, in which case we can re-open it.

**Comment by MemeCornucopia** [comment]
Well actually I did a little digging, I downloaded their mobile app which is for "premium" users only, but I did a little trolling and intercepted the requests from the server and told the client I had premium I then found out that they have no captchas / protection for their mobile api thus making this actually possible

**Comment by MemeCornucopia** [comment]
image
image

https://lmscript.xyz/v1/movies/view?expand=streams&id=110465

**Comment by mrjvs** [comment]
From the looks of it, it looks possible. No accounts, no cors protection, no captchas.

Who wants to pick this up?

**Pull request closed without merging**